utils/directories.cc

```diff
@@ -140,7 +148,23 @@ future<> directories::do_verify_owner_and_mode(fs::path path, recursive recurse,
 };
 
 future<> directories::verify_owner_and_mode(fs::path path, recursive recursive) {
-    return do_verify_owner_and_mode(std::move(path), recursive, 0);
+    return do_verify_owner_and_mode(std::move(path), recursive, 0, [](fs::path &path){return true;});
+}
+
+future<> directories::verify_owner_and_mode_of_data_dir(directories::set dir_set) {
+    co_await parallel_for_each(dir_set.get_paths(), [](const auto &path) -> future<> {
+        co_await do_verify_owner_and_mode(std::move(path), recursive::yes, 0, [](fs::path &path) {
+            const sstring &file_name = path.filename().string();
```

Comment:
nit: coding style: & sticks to type.
also, you can use auto&

And also elsewhere we use native() to get to the path in a native format string  rather than string()

Reply:
Done.